src/Facets/SquidFacet.sol

```diff
@@ -0,0 +1,133 @@
+// SPDX-License-Identifier: MIT
+pragma solidity 0.8.17;
+
+import { ILiFi } from "../Interfaces/ILiFi.sol";
+import { ISquidRouter } from "../Interfaces/ISquidRouter.sol";
+import { ISquidMulticall } from "../Interfaces/ISquidMulticall.sol";
+import { LibAsset, IERC20 } from "../Libraries/LibAsset.sol";
+import { LibSwap } from "../Libraries/LibSwap.sol";
+import { LibUtil } from "../Libraries/LibUtil.sol";
+import { ReentrancyGuard } from "../Helpers/ReentrancyGuard.sol";
+import { SwapperV2 } from "../Helpers/SwapperV2.sol";
+import { Validatable } from "../Helpers/Validatable.sol";
+
+/// @title Squid Facet
+/// @author LI.FI (https://li.fi)
+/// @notice Provides functionality for bridging through Squid Router
+contract SquidFacet is ILiFi, ReentrancyGuard, SwapperV2, Validatable {
+    enum SquidCallType {
+        BridgeCall,
+        CallBridge,
+        CallBridgeCall
+    }
+
+    struct SquidData {
+        SquidCallType callType;
+        bytes callData;
+    }
```

Comment:
Instead of passing callData I would use a type similar to `callBridgeCall`, but pass token, amount, destinationAddress from the bridge data. We want to be able to use as much of the bridgeData as possible, to be able to rely on that that, e.g. to validate what will happen or to adjust the amount without knowing the full callData.

Reply:
```
    function bridgeCall(
        string calldata destinationChain,
        string calldata bridgedTokenSymbol,
        uint256 amount, <= _bridgeData.minAmount
        ISquidMulticall.Call[] calldata calls,
        address refundRecipient, <= _bridgeData.recipient
        bool forecallEnabled
    ) external payable;

    function callBridge(
        address token, <= _bridgeData.sendingAssetId
        uint256 amount, <= _bridgeData.minAmount
        string calldata destinationChain,
        string calldata destinationAddress, <= _bridgeData.recipient
        string calldata bridgedTokenSymbol,
        ISquidMulticall.Call[] calldata calls
    ) external payable;

    function callBridgeCall(
        address token, <= _bridgeData.sendingAssetId
        uint256 amount, <= _bridgeData.minAmount
        string calldata destinationChain,
        string calldata bridgedTokenSymbol,
        ISquidMulticall.Call[] calldata sourceCalls,
        ISquidMulticall.Call[] calldata destinationCalls,
        address refundRecipient, <= _bridgeData.recipient
        bool forecallEnabled
    ) external payable;
```

Reply:
The problem is it is very difficult to rebuild the Call[] array from the information passed by the API alone.